### PR TITLE
try mounting without fusermount(1) first

### PR DESCRIPTION
**Pull request by stapelberg**
This requires root privileges or the CAP_SYS_ADMIN capability, but allows supporting -o suid.

I also addressed a bunch of readability-related nits while touching the code.

### CODE REVIEW

File: mount_linux.go [hunk 1]
```diff
@@ -126,10 +126,11 @@ var mountflagopts = map[string]func(uintptr) uintptr{
 var errFallback = errors.New("sentinel: fallback to fusermount(1)")
 
 func directmount(dir string, cfg *MountConfig) (*os.File, error) {
-	dev, err := os.OpenFile("/dev/fuse", os.O_RDWR, 0644)
+	fd, err := syscall.Open("/dev/fuse", syscall.O_RDWR, 0644)
```
Comment (hugelgupf) [inline mount_linux.go]: I'd also recommend a comment here saying why this isn't an os.OpenFile call

Comment (stapelberg) [reply mount_linux.go]: Done.

**Comment by kungf** [comment]
hi, @stapelberg  can you help me,with this patch when i mount fuse with root, hit the err
"mount: invalid argument"
seems mount source not work, can you give some advise?

but normal user is ok!

**Comment by stapelberg** [comment]
Can you open an issue for that please? In the issue, can you supply an strace log (use strace -fvy -s2048 -o /tmp/strace.log in front of your command), and can you let us know how to reproduce the problem?

**Comment by kungf** [comment]
@stapelberg 

> Can you open an issue for that please? In the issue, can you supply an strace log (use strace -fvy -s2048 -o /tmp/strace.log in front of your command), and can you let us know how to reproduce the problem?

https://github.com/jacobsa/fuse/issues/76

thank you very much!

**Comment by kungf** [comment]
@stapelberg hi， can you tell something about why root not use fusermount？

**Comment by stapelberg** [comment]
This is a requirement for supporting the `-o suid` option. Also, it helps in early-boot environments where fusermount might not be available.

**Comment by kungf** [comment]
> This is a requirement for supporting the `-o suid` option. Also, it helps in early-boot environments where fusermount might not be available.

got it, thank you

**Pull request merged**